xml/src/main/scala/akka/stream/alpakka/xml/impl/StreamingXmlParser.scala

```diff
@@ -58,67 +62,72 @@ private[xml] object StreamingXmlParser {
         else if (isAvailable(out)) advanceParser()
       }
 
-      @tailrec private def advanceParser(): Unit =
+       private def advanceParser(): Unit = {
```

Comment:
Scala `tailrec` does not like exceptions. I think the exception handling overrides the annotation but am open to other opinions. This could also be converted to a `while` if really wanted

Reply:
Pull out the try/catch to the graph stage API methods invoking it instead and leave this as it was?

Reply:
Yah - I guess I was trying to centralize and got caught up in that. I just pushed a change with that suggestion